contracts/IERC721M.sol

```diff
@@ -26,13 +26,17 @@ interface IERC721M is IERC721AQueryable {
     error WalletStageLimitExceeded();
     error WithdrawFailed();
 
+    // The type of the sale
+    enum SaleType { DirectSale, BucketAuction }
```

Comment:
this is a leaky abstraction from the more generic contract - I think in general, we probably want the ERC721M to be the base to all the extension contracts, such that it's composable instead of inheritance to define the possible values of all the sale type.

- it adds complexity to the vanila smart contract that now we need to specify saleType for all stages
- adding new sale type requiring bumping the base smart contract npm version. which better to be done purely in the extension contract.

Wondering if we can just change BucketAuction.sol for this PR without adding the saleType, so that the change of determining if bucketAuction is live or not is self contained - i.e. we replace setActive with start/end time in BucketAuction.sol